src/vs/platform/log/electron-main/logIpc.ts

```diff
@@ -0,0 +1,71 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the MIT License. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+import { Event } from 'vs/base/common/event';
+import { URI } from 'vs/base/common/uri';
+import { IServerChannel } from 'vs/base/parts/ipc/common/ipc';
+import { ILogger, ILoggerOptions, log, LogLevel } from 'vs/platform/log/common/log';
+import { ILoggerMainService } from 'vs/platform/log/electron-main/loggerService';
+
+export class LoggerChannel implements IServerChannel {
+
+	private readonly loggers = new Map<string, ILogger>();
+
+	constructor(private readonly loggerService: ILoggerMainService) { }
+
+	listen(_: unknown, event: string, windowId?: number): Event<any> {
+		switch (event) {
+			case 'onDidChangeLoggerResources': return windowId ? this.loggerService.getOnDidChangeLoggerResourcesEvent(windowId) : this.loggerService.onDidChangeLoggerResources;
+			case 'onDidChangeLogLevel': return windowId ? this.loggerService.getOnDidChangeLogLevelEvent(windowId) : this.loggerService.onDidChangeLogLevel;
+		}
+		throw new Error(`Event not found: ${event}`);
+	}
+
+	async call(_: unknown, command: string, arg?: any): Promise<any> {
+		switch (command) {
+			case 'createLogger': this.createLogger(URI.revive(arg[0]), arg[1]); return;
```

Comment:
Why not use our automated proxy services?

Reply:
Because, this has some special handling for following methods and events

- `createLogger`
- `log`
- `consoleLog`
- `onDidChangeLoggerResources`
- `onDidChangeLogLevel`

Also, some of these methods are not defined in the service. Not sure if proxy services support this.